dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -395,7 +400,16 @@ Path findClassLocation(String className) {
         javaClasses.add(javaClass);
       } catch (ClassNotFoundException ex) {
         // We couldn't find the class in the jar file where we found it.
-        throw new IOException("Corrupt jar file " + jar + "; could not load " + classFileName, ex);
+        Throwable cause = ex.getCause();
+        if (cause != null && cause instanceof IOException) {
+          // Skip unexpected files (such as a lock file) included in JAR file
+          // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1048 was caused
+          // by an IOException
+          corruptedClassFileNames.add(classFileName);
+        } else {
+          throw new IOException(
+              "Corrupt jar file " + jar + "; could not load " + classFileName, ex);
```

Comment:
so we'll still die on these sorts of corrupt jar files? Is that what we want? Perhaps whoever calls this method should handle the IOExceptions that arise here.

Reply:
That's a nice idea. Now `findSymbolReferences` catches the exception.

Reply:
I found the problem of catching IOException at the caller; it would skip all _valid_ classes in the JAR file.